identity-apps-core/apps/authentication-portal/src/main/webapp/org_discovery.jsp

```diff
@@ -43,13 +44,17 @@
            errorMessage = request.getParameter(Constants.AUTH_FAILURE_MSG);
            if (errorMessage.equalsIgnoreCase("authentication.fail.message")) {
                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "error.retry");
-           } else if (errorMessage.equalsIgnoreCase("Can't identify organization")) {
-                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "invalid.organization.discovery.input");
-           } else if (errorMessage.equalsIgnoreCase("invalid.organization.discovery.type")) {
+            } else if (isSelfRegistration && errorMessage.equalsIgnoreCase("Can't identify organization")) {
+               errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "invalid.organization.discovery.input.self.registration");
+            } else if (errorMessage.equalsIgnoreCase("Can't identify organization")) {
+               errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "invalid.organization.discovery.input");
+            } else if (errorMessage.equalsIgnoreCase("invalid.organization.discovery.type")) {
                errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle, "invalid.organization.discovery.type");
-           } else if (isErrorFallbackLocale) {
-               errorMessage = AuthenticationEndpointUtil.i18n(resourceBundle,"error.retry");
-           }
+            } else if (isSelfRegistration && errorMessage.equalsIgnoreCase("Organization is not associated with this application.")) {
```

Comment:
use a key instead of this string

Reply:
This is the error message currently returned from the organization SSO authenticator. Since this is the existing behaviour will track this improvement as a separate effort.